packages/gatsby-transformer-hjson/src/gatsby-node.js

```diff
@@ -12,7 +14,7 @@ async function onCreateNode({ node, actions, loadNodeContent }) {
       .digest(`hex`)
     const jsonNode = {
       ...obj,
-      id,
+      id: createNodeId(id),
```

Comment:
Looks like we never were creating an ID if it was missing. Could you add that? But same here as the others, don't hash the manually created ID